### PR TITLE
[Method References] LPS-161456 Convert method references to field references in module portal-search-tuning

### PR DESCRIPTION
**Pull request by W2jx**
Hi @tinatian @dantewang , I convert method references to field references and fix tests in this pr.

### CODE REVIEW

**Comment by W2jx** [comment]
Hi @zhangyan0701 , Could you please help me check the test results when they finish ? Thanks!

**Comment by zhangyan0701** [comment]
ci:test:search

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/index/SynonymSetIndexReaderImpl.java [hunk 1]
```diff
@@ -78,13 +78,6 @@ public List<SynonymSet> search(SynonymSetIndexName synonymSetIndexName) {
 			searchSearchResponse.getSearchHits());
 	}
 
-	@Reference(unbind = "-")
-	protected void setSearchEngineAdapter(
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-synonyms-web/src/main/java/com/liferay/portal/search/tuning/synonyms/web/internal/index/SynonymSetIndexReaderImpl.java:82]: @W2jx This change should be put in a separate commit "remove duplicate reference", or I would prefer the word "redundant" over "duplicate" because the field is already referenced.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java [hunk 2]
```diff
@@ -34,23 +35,25 @@ public class RankingIndexNameBuilderImplTest {
...
-	}
-
-	private void _assertIndexName(
-		long companyId, IndexNameBuilder indexNameBuilder, String expected) {
-
 		RankingIndexNameBuilderImpl rankingIndexNameBuilderImpl =
 			new RankingIndexNameBuilderImpl();
 
-		rankingIndexNameBuilderImpl.setIndexNameBuilder(indexNameBuilder);
+		ReflectionTestUtil.setFieldValue(
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java:41]: @W2jx In this commit, you fixed the usage in `_assertIndexName` and then inlined `_assertIndexName` into `testMultiTenancy`. The former change is "fix test" while the latter change is "SF". So you should do it in 2 commits.

**Review by dantewang — CHANGES_REQUESTED** [review]
@W2jx Please see my comments and update the pull.

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java [hunk 3]
```diff
@@ -35,24 +35,25 @@ public class RankingIndexNameBuilderImplTest {
...
-	private void _assertIndexName(
-		long companyId, IndexNameBuilder indexNameBuilder, String expected) {
-
 		RankingIndexNameBuilderImpl rankingIndexNameBuilderImpl =
 			new RankingIndexNameBuilderImpl();
 
 		ReflectionTestUtil.setFieldValue(
-			rankingIndexNameBuilderImpl, "_indexNameBuilder", indexNameBuilder);
+			rankingIndexNameBuilderImpl, "_indexNameBuilder",
+			new IndexNameBuilder() {
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java:43]: @W2jx Another thing, the original code creates the `IndexNameBuilder` using `companyId -> "liferay-" + companyId`, which is a lambda, why change it back to anonymous class?

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java [hunk 4]
```diff
@@ -35,24 +35,25 @@ public class RankingIndexNameBuilderImplTest {
...
+				}
 
-		RankingIndexName rankingIndexName =
-			rankingIndexNameBuilderImpl.getRankingIndexName(companyId);
+			});
 
-		Assert.assertEquals(expected, rankingIndexName.getIndexName());
+		Assert.assertEquals(
+			"liferay-2021-search-tuning-rankings",
+			rankingIndexNameBuilderImpl.getRankingIndexName(
```
Comment (dantewang) [inline modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java:54]: @W2jx Please don't inline this, basically we don't allow chaining methods like `A.doOneThing().doAnotherThing()`.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@W2jx Please see my comment, fix the SF issues, and resend.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java (hunk 3 above)
Comment (W2jx) [reply modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java:43]: Because input parameters of method setFieldValue are (Object, string, Object), and lambda cannot be assigned to Object.

File: modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java (hunk 4 above)
Comment (W2jx) [reply modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web/src/test/java/com/liferay/portal/search/tuning/rankings/web/internal/index/name/RankingIndexNameBuilderImplTest.java:54]: Ok, I will fix it.
